enhancements/proposals/1665-metadata/README.md

```diff
@@ -0,0 +1,122 @@
+
+### Non-Goals
+
+- Allow unique metadata for every single object we create 
+
+## Proposal
+
+I'd like to use the existing ScyllaCluster.metadata (labels and annotations) and inherit them in every object we create,
+unless there is an override.
+In the case when using an override, the ScyllaCluster.metadata won't be used.
+
+I propose to extend the ScyllaCluster API to allow overriding metadata for pods, node services and ingresses.
+
+In case there is a conflict with managed metadata, the managed metadata will win.
+
+These changes are purely intended for metadata and shall not change the selectors used.
+
+### User Stories
+
+#### DataDog metrics
+As a user I want to configure [DataDog metrics](https://docs.datadoghq.com/containers/kubernetes/prometheus/?tab=kubernetesadv2#setup) for certain pods.
+This is done by adding `ad.datadoghq.com/<CONTAINER_IDENTIFIER>.checks` to Pod's annotations.
+
+#### Third Party Service and Ingress options
+As a user I want to configure additional options for my service or ingress.
+These third party options are usually controlled by adding annotations or labels, e.g.:
+- [`networking.gke.io/load-balancer-type: "Internal"` annotation for GKE](https://cloud.google.com/kubernetes-engine/docs/how-to/internal-load-balancing)
+- [`service.beta.kubernetes.io/aws-load-balancer-eip-allocations: eipalloc-xxxxxxxxxxxxxxxxx,eipalloc-yyyyyyyyyyyyyyyyy` annotation for assigning elastic IPs](https://docs.aws.amazon.com/eks/latest/userguide/network-load-balancing.html)
+- [Ingress Controller sharding using labels and selectors in OCP](https://docs.openshift.com/container-platform/4.14/networking/ingress-sharding.html)
+
+#### Cost tracking
+As a user I want to add a label to all of my resources and use it to track costs.
+
+#### PagerDuty
+As a user I want to add extra label to my workload to route dynamic PagerDuty alerts.
+
+### Risks and Mitigations
+
+Not known.
+
+## Design Details
+
+### API changes for ScyllaClusters.scylla.scylladb.com/v1
+
+The following API changes show how we'd approximately extend `scyllaclusters.scylla.scylladb.com/v1.spec` specification. Only new fields are shown for existing structs.
+
+```golang
+package v1
+
+type ObjectTemplateMetadata struct {
+	// labels is a custom key value map that gets merged with managed object annotations.
+	// +optional
+	Labels map[string]string `json:"labels,omitempty"`
+
+	// annotations is a custom key value map that gets merged with managed object annotations.
+	// +optional
+	Annotations map[string]string `json:"annotations,omitempty"`
+}
+
+type ScyllaClusterSpec struct {
+	// podMetadata controls shared metadata for all pods created based on this spec.
+	// +optional
+	PodMetadata *ObjectTemplateMetadata `json:"podMetadata,omitempty"`
+
+	// ...
+}
+
+type IngressOptions struct {
+	ObjectTemplateMetadata `json:",inline"`
+	
+	// ...
+}
+
+
+type NodeServiceTemplate struct {
+    // ObjectTemplateMetadata "replaces" / embeds Labels
```

Comment:
Looks like a leftover?

Reply:
it's only for the purposes of this proposal to show that this removes a field named `annotations` (I'll fix the typo)
see https://github.com/scylladb/scylla-operator/pull/1664/files#diff-73814bffab13ea418b67e2c20f449963e2240a1a07ce288ca3c5919fccf10272L185

Reply:
I see, thanks.